### PR TITLE
Added dtos for movie, movieSession,user and controllers for this entities

### PR DESCRIPTION
**Pull request by andriylagun**

### CODE REVIEW

File: src/main/java/com/online/cinema/dto/movie/MovieResponseDto.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.online.cinema.dto.movie;
+
+import lombok.Data;
+
+@Data
+public class MovieResponseDto {
+    private Long id;
+    private String title;
```
Comment (AnaLanda) [inline src/main/java/com/online/cinema/dto/movie/MovieResponseDto.java]: Maybe we'll need to have the `description` here as well

File: src/main/java/com/online/cinema/dto/moviesession/MovieSessionRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.online.cinema.dto.moviesession;
+
+import lombok.Data;
+
+@Data
+public class MovieSessionRequestDto {
+    private Long movieId;
+    private Long cinemaHallId;
+    private String sessionTime;
```
Comment (AnaLanda) [inline src/main/java/com/online/cinema/dto/moviesession/MovieSessionRequestDto.java]: I think you can have a String that represents date+time here, like the `LocalDateTime` format `ISO_LOCAL_DATE_TIME` or something like it

File: src/main/java/com/online/cinema/dto/moviesession/MovieSessionResponseDto.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+import lombok.Data;
+
+@Data
+public class MovieSessionResponseDto {
+    private Long id;
+    private Long movieId;
+    private Long cinemaHallId;
+    private String sessionDate;
+    private String sessionTime;
```
Comment (AnaLanda) [inline src/main/java/com/online/cinema/dto/moviesession/MovieSessionResponseDto.java]: Same here

File: src/main/java/com/online/cinema/dto/user/UserRegisterDto.java [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package com.online.cinema.dto.user;
+
+import lombok.Data;
+
+@Data
+public class UserRegisterDto {
```
Comment (AnaLanda) [inline src/main/java/com/online/cinema/dto/user/UserRegisterDto.java]:
```suggestion
public class UserRequestDto {
```

**Review by AnaLanda — COMMENTED** [review]
Looks great! 😃

File: src/main/java/com/online/cinema/controller/AuthenticationController.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+public class AuthenticationController {
+    private AuthenticationService authenticationService;
+
+    @Autowired
+    public AuthenticationController(AuthenticationService authenticationService) {
+        this.authenticationService = authenticationService;
+    }
+
+    @PostMapping("/registration")
+    private RedirectView register(@RequestBody UserRegisterDto userDto) {
```
Comment (Sofasmile) [inline src/main/java/com/online/cinema/controller/AuthenticationController.java]: зараз ми можемо зробити цей метод void, а в майбутньому повертати статус код що операція пройшла успішно

File: src/main/java/com/online/cinema/controller/MovieController.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    @GetMapping
+    public List<MovieResponseDto> getAllMovies() {
+        return movieService.getAll().stream()
+                .map(movieDtoMapper::toResponseDto)
+                .collect(Collectors.toList());
+    }
+
+    @PostMapping
+    public RedirectView addMovie(@RequestBody MovieRequestDto movieRequestDto) {
```
Comment (Sofasmile) [inline src/main/java/com/online/cinema/controller/MovieController.java]: the same here

File: src/main/java/com/online/cinema/controller/MovieSessionController.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+            MovieSessionDtoMapper movieSessionDtoMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionDtoMapper = movieSessionDtoMapper;
+    }
+
+    @PostMapping
+    public RedirectView addMovieSession(
+            @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.add(movieSessionDtoMapper.fromRequestDto(movieSessionRequestDto));
+        return new RedirectView("/avaliable");
```
Comment (Sofasmile) [inline src/main/java/com/online/cinema/controller/MovieSessionController.java]: and here

File: src/main/java/com/online/cinema/controller/MovieSessionController.java [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAllAvailableSessions(Long movieId, String date) {
```
Comment (Sofasmile) [inline src/main/java/com/online/cinema/controller/MovieSessionController.java]: let's change String to LocalDate

File: src/main/java/com/online/cinema/config/AppConfig.java [hunk 9]
```diff
@@ -15,7 +15,8 @@
 @ComponentScan(basePackages = {
         "com.online.cinema.service",
         "com.online.cinema.dao",
-        "com.online.cinema.security"
+        "com.online.cinema.security",
+        "com.online.cinema.dto"
```
Comment (Sofasmile) [inline src/main/java/com/online/cinema/config/AppConfig.java]:
```suggestion
        "com.online.cinema.mapper or com.online.cinema.dto.mapper"
```

File: src/main/java/com/online/cinema/dto/moviesession/MovieSessionRequestDto.java (hunk 2 above)
Comment (Sofasmile) [inline src/main/java/com/online/cinema/dto/moviesession/MovieSessionRequestDto.java]:
```suggestion
    private String sessionDateTime;
```

File: src/main/java/com/online/cinema/dto/moviesession/MovieSessionResponseDto.java (hunk 3 above)
Comment (Sofasmile) [inline src/main/java/com/online/cinema/dto/moviesession/MovieSessionResponseDto.java]:
```suggestion
    private String sessionDateTime;
```

File: src/main/java/com/online/cinema/controller/MovieSessionController.java [hunk 10]
```diff
@@ -29,15 +28,14 @@ public MovieSessionController(MovieSessionService movieSessionService,
...
+    public void addMovieSession(
             @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
         movieSessionService.add(movieSessionDtoMapper.fromRequestDto(movieSessionRequestDto));
-        return new RedirectView("/avaliable");
     }
 
     @GetMapping("/available")
-    public List<MovieSessionResponseDto> getAllAvailableSessions(Long movieId, String date) {
-        return movieSessionService.findAvailableSessions(movieId, LocalDate.parse(date))
+    public List<MovieSessionResponseDto> getAllAvailableSessions(Long movieId, LocalDate date) {
```
Comment (Sofasmile) [inline src/main/java/com/online/cinema/controller/MovieSessionController.java]: does this method work correctly?

File: src/main/java/com/online/cinema/controller/MovieSessionController.java [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+    @PostMapping
+    public void addMovieSession(
+            @RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.add(movieSessionDtoMapper.fromRequestDto(movieSessionRequestDto));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAllAvailableSessions(
+            @RequestParam Long movieId, @RequestParam LocalDate date) {
```
Comment (KseniiaMakarova) [inline src/main/java/com/online/cinema/controller/MovieSessionController.java]: please provide the screenshot if this works

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**